cpu/nrf52/radio/nrf802154/nrf802154.c

```diff
@@ -247,6 +248,11 @@ static int _init(netdev_t *dev)
     txbuf[0] = 0;
     _state = 0;
 
+    /* the radio need the external HF clock source to be enabled */
+    /* @todo    add proper handling to release the clock whenever the radio is
+     *          idle */
+    clock_hfxo_request();
```

Comment:
I think `nrf802154_radio.c` will need an update too. This always needs to be updated together with `NRF_RADIO->POWER`, right?

Reply:
done

Reply:
Since those always occur together, how about adding a 

```C
static inline void nrf5x_radio_on(void)
{
    clock_hfxo_request();
    NRF_RADIO->POWER = 1;
}

static inline void nrf5x_radio_off(void)
{
    NRF_RADIO->POWER = 0;
    clock_hfxo_release();
}
```

Reply:
borderline benefit for those 3 occurrences I'd say :-)

Reply:
Well it has the advantage that you don't miss any.
Just replace `NRF_RADIO->POWER = 0` with `nrf5x_radio_off()` and `NRF_RADIO->POWER = 1` with `nrf5x_radio_on()`  and you know you are safe if the POWER register is not modified manually anymore.

Reply:
done. But did this using private functions, as I think exposing these to the global API might have implications which are potentially breaking the radios state.